### PR TITLE
Move single-use symbols out of pkg/common

### PR DESCRIPTION
**Pull request by tklauser**
Packages like _common_ or _util_ usually become a dumping ground for all sorts of unrelated code, see e.g. https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common

Thus, slim down `pkg/common` a bit by moving funcs and consts used only by a single package to that respective package and also un-export them.

### CODE REVIEW

**Comment by tklauser** [comment]
test-me-please

**Review by rolinh — APPROVED** [review]

File: pkg/endpoint/directory_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	oldDir := c.MkDir()
+	newDir := c.MkDir()
+	f1, err := ioutil.TempFile(oldDir, "")
+	c.Assert(err, check.IsNil)
+	f2, err := ioutil.TempFile(oldDir, "")
+	c.Assert(err, check.IsNil)
+	f3, err := ioutil.TempFile(newDir, "")
+	c.Assert(err, check.IsNil)
+
+	// Copy the same f4 file in both directories to make sure the same files
```
Comment (kaworu) [inline pkg/endpoint/directory_test.go]: typo?
```suggestion
	// Copy the same file in both directories to make sure the same files
```

**Review by kaworu — APPROVED** [review]

File: pkg/endpoint/directory_test.go (hunk 1 above)
Comment (tklauser) [reply pkg/endpoint/directory_test.go]: The comment was already there in `pkg/common` and it looks like it refers to an `f4 *os.File` which never existed. Will fix.

**Comment by tklauser** [comment]
test-me-please

**Review by christarazi — APPROVED** [review]
Nice!

**Review by aanm — APPROVED** [review]

**Pull request merged**